File: nestjs/src/hello/hello.controller.ts
import { Controller, Get, Post, Body } from "@nestjs/common";
import { HelloService } from "./hello.service";

@Controller("api/hello")
export class HelloController {
  constructor(private readonly helloService: HelloService) {}

  @Get()
  getHello() {
    return this.helloService.getHello();
  }

  @Post()
  greet(@Body("name") name: string) {
    return this.helloService.greet(name);
  }
}

File: spring-boot/src/main/java/com/example/app/controller/HelloController.java
package com.example.app.controller;

import com.example.app.model.HelloRequest;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/hello")
@CrossOrigin(origins = "*")
public class HelloController {

    @GetMapping
    public ResponseEntity<Map<String, String>> getHello() {
        return ResponseEntity.ok(Map.of("message", "Hello from Spring Boot!"));
    }

    @PostMapping
    public ResponseEntity<Map<String, String>> postHello(@Valid @RequestBody HelloRequest req) {
        return ResponseEntity.ok(Map.of("message", "Hello, " + req.name() + "!"));
    }
}
